endpoints/users_id_token.py

```diff
@@ -327,27 +346,27 @@ def _set_bearer_user_vars(allowed_client_ids, scopes):
     allowed_client_ids: List of client IDs that are acceptable.
     scopes: List of acceptable scopes.
   """
-  for scope in scopes:
-    try:
-      client_id = oauth.get_client_id(scope)
-    except oauth.Error:
-      # This scope failed.  Try the next.
-      continue
-
-    # The client ID must be in allowed_client_ids.  If allowed_client_ids is
-    # empty, don't allow any client ID.  If allowed_client_ids is set to
-    # SKIP_CLIENT_ID_CHECK, all client IDs will be allowed.
-    if (list(allowed_client_ids) != SKIP_CLIENT_ID_CHECK and
-        client_id not in allowed_client_ids):
-      _logger.warning('Client ID is not allowed: %s', client_id)
-      return
+  all_scopes, sufficient_scopes = _process_scopes(scopes)
+  try:
+    authorized_scopes = oauth.get_authorized_scopes(sorted(all_scopes))
+  except oauth.Error:
+    _logger.debug('Unable to get authorized scopes.', exc_info=True)
+    return
+  if not _are_scopes_sufficient(authorized_scopes, sufficient_scopes):
+    _logger.debug('Authorized scopes did not satisfy scope requirements.')
```

Comment:
Shouldn't we return or fail here?

Also, would it be useful to include the actual authorized/sufficient scopes in the log message?

Reply:
Fair enough